src/main/ApplicationImpl.cpp

```diff
@@ -609,6 +609,39 @@ ApplicationImpl::syncAllMetrics()
     syncOwnMetrics();
 }
 
+template <typename T>
+std::shared_ptr<T>
+castMetric(std::shared_ptr<medida::MetricInterface> metric)
+{
+    return std::dynamic_pointer_cast<T>(metric);
+}
+
+void
+ApplicationImpl::clearMetrics(std::string domain)
+{
+    auto const& metrics = mMetrics->GetAllMetrics();
+    for (auto const& kv : metrics)
+    {
+        auto metricName = kv.first;
+        if (domain.empty() || metricName.domain() == domain)
+        {
+            auto metric = kv.second;
+            if (auto counter = castMetric<medida::Counter>(metric))
+            {
+                counter->clear();
```

Comment:
are the clear methods thread safe?

Reply:
Looks like it. Timer and Histogram are using mutex, while counter only clears one variable and uses `std::atomic`.